src/main/java/net/querz/nbt/ListTag.java

```diff
@@ -242,131 +266,135 @@ public ListTag getList(int index) {
 		return (ListTag) value.get(index);
 	}
 
+	public Number getNumber(int index) {
+		return getNumberTag(index).asNumber();
+	}
+
 	public boolean getBoolean(int index) {
 		return getByte(index) != 0;
 	}
 
-	public byte getByteOrDefault(int index, byte def) {
-		if (index >= 0 && index < value.size()) {
-			Tag tag = value.get(index);
-			if (tag instanceof NumberTag) {
-				return ((NumberTag) tag).asByte();
-			}
+	public Tag getTagOrNull(int index) {
+		if (index < value.size()) {
```

Comment:
this is missing the check that index is >= 0

Reply:
I had this initially, but decided that if you're passing in a negative index something must've gone so horribly wrong somewhere that it's worth failing fast. This is consistent with other libraries like guava's [`Iterables.get`](https://javadoc.io/static/com.google.guava/guava/32.0.0-jre/com/google/common/collect/Iterables.html#get(java.lang.Iterable,int,T)).

Reply:
Sorry I've been gone for a while, how do you feel about this now @Querz?